popsycle/synthetic.py

```diff
@@ -3146,3 +3154,454 @@ def calc_f(lambda_eff):
+        Inner_slope goes into the determination of the velocities and each value returns a slightly different distribution.
+        The default value is .5 because it is in the middle of the options. More investigation is needed.
+
+    v_esc: int
+        The escape velocity of the Milky Way.
+        v_esc is used in calculating the velocities.
+        Default is 550 because most papers cite values of 515-575, with a lot being around 550.
+
+    overwrite : bool
+        If set to True, overwrites output files. If set to False, exists the
+        function if output files are already on disk.
+        Default is False.
+
+    seed : int
+        If set to non-None, all random sampling will be seeded with the
+        specified seed, forcing identical output for PyPopStar and PopSyCLE.
+        Default None.
+
+    Outputs
+    -------
+    <output_root2>.h5 : hdf5 file
+        The new .h5 file with PBHs injected in.
+    """
+    ##########
+    # Error handling: check whether files exist and
+    # whether input types are correct.
+    ##########
+
+    if not overwrite:
+        # Check if HDF5 file exists already. If it does, throw an error message
+        # to complain and exit.
+        if os.path.isfile(output_root2 + '.h5'):
+            raise Exception(
+                'That .h5 file name is taken! Either delete the .h5 file, '
+                'or pick a new name.')
+
+    # Error handling/complaining if input types are not right.
+    if ebf_file[-4:] != '.ebf':
+        raise Exception('ebf_file must be an ebf file.')
+
+    if type(output_root2) != str:
+        raise Exception('output_root must be a string.')
+
+    if type(fdm) != float:
+        if type(fdm) != int:
+            raise Exception('fdm must be a float or an integer.')
+
+    if type(pbh_mass) != int:
+        if type(pbh_mass) != float:
+            raise Exception('pbh_mass must be an integer or a float.')
+
+    if type(r_max) != float:
+        if type(r_max) != int:
+            raise Exception('r_max must be a float or an integer.')
+
+    if type(c) != int:
+        if type(c) != float:
+            raise Exception('c must be an integer or a float.')
+
+    if type(r_vir) != float:
+        if type(r_vir) != int:
+            raise Exception('r_vir must be a float or an integer.')
+
+    if seed is not None:
+        if type(seed) != int:
+            raise Exception('seed must be an integer.')
+
+    # Check to make sure that the output hdf5 file
+    # will not overwrite the input hdf5 file
+    output_hdf5_file = '%s.h5' % output_root2
+    if hdf5_file == output_hdf5_file:
+        raise Exception('Output hdf5 file %s cannot be equal to '
+                        'input hdf5 file %s' % (output_hdf5_file, hdf5_file))
+
+    ##########
+    # Start of code
+    #########
+
+    # Set random seed
+    np.random.seed(seed)
+
+    t0 = time.time()
+
+    #Define parameters for NFW profile calculations
```

Comment:
I don't think we need these three parameters anymore given the new model.

Reply:
Just to clarify. I am suggesting that we take r_s as input to the function rather than r_vir and c. 1) it is what is fit in McMillan (17), 2) it is one parameter instead of 2, 3) r_vir and c aren't used elsewhere.